pyintegration/deephaven2/time.py

```diff
@@ -0,0 +1,830 @@
+        DateTime
+
+    Raises:
+        DHError
+    """
+    try:
+
+        return _JDateTimeUtils.dateAtMidnight(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def day_of_month(dt: DateTime, tz: TimeZone) -> int:
+    """ Returns an 1-based int value of the day of the month for a DateTime and specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the day of the month
+        tz (TimeZone): the TimeZone to use when interpreting the DateTime
+
+    Returns:
+        int: NULL_INT if dt is None
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.dayOfMonth(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def day_of_week(dt: DateTime, tz: TimeZone) -> int:
+    """ Returns an 1-based int value of the day of the week for a DateTime in the specified time zone, with 1 being
+     Monday and 7 being Sunday.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the day of the week.
+        tz (TimeZone): the TimeZone to use when interpreting the Datetime.
+
+    Returns:
+        int: NULL_INT if dt is None
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.dayOfWeek(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def day_of_year(dt: DateTime, tz: TimeZone) -> int:
+    """ Returns an 1-based int value of the day of the year (Julian date) for a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the day of the year
+        tz (TimeZone): the TimeZone to use when interpreting the DateTime
+
+    Returns:
+        int: NULL_INT if dt is None
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.dayOfYear(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def diff_nanos(dt1: DateTime, dt2: DateTime) -> int:
+    """ Returns the difference in nanoseconds between two DateTime values.
+
+    Args:
+        dt1 (DateTime): the 1st DateTime
+        dt2 (DateTime): the 2nd DateTime
+
+    Returns:
+        int: NULL_LONG if either dt1 or dt2 is None
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.diffNanos(dt1, dt2)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def format_datetime(dt: DateTime, tz: TimeZone) -> str:
+    """ Returns a string DateTime representation formatted as "yyyy-MM-ddThh:mm:ss.SSSSSSSSS TZ".
+
+    Args:
+        dt (DateTime): the DateTime to format as a string
+        tz (TimeZone): the TimeZone to use when interpreting the DateTime
+
+    Returns:
+        str
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.format(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def format_nanos(ns: int) -> str:
+    """ Returns a string DateTime representation formatted as "yyyy-MM-ddThh:mm:ss.SSSSSSSSS".
```

Comment:
Is this formatting epoch nanos to a string date time?  If so, it should take in a timezone and output a full "yyyy-MM-ddThh:mm:ss.SSSSSSSSS TZ" format.  On the other hand, if this takes in nanos like 1*HOUR+2*MINUTE+3*SECOND and returns "hh:mm:ss.SSSSSSSSS", then the function signature is fine, but the documentation isn't.

Reply:
yes, the doc is wrong.